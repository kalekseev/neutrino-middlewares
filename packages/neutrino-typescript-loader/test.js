import { Neutrino } from 'neutrino';

// eslint-disable-next-line global-require, import/no-unresolved
const mw = () => require('.');

test('loads preset', () => {
  mw();
});

test('uses preset', () => {
  Neutrino().use(mw());
});

test('valid preset production', () => {
  const api = Neutrino({ env: { NODE_ENV: 'production' }, command: 'build' });
  api.use(mw());
  const config = api.config.toConfig();
  expect(config).toMatchSnapshot();
});

test('valid preset development', () => {
  const api = Neutrino({ 'env': { NODE_ENV: 'development' }, command: 'start' });
  api.use(mw());
  const config = api.config.toConfig();
  expect(config).toMatchSnapshot();
});
