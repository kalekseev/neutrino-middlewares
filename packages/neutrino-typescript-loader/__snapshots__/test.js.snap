// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`valid preset development 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\\\\\.tsx\\?\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/awesome-typescript-loader/dist/entry.js",
          },
        ],
      },
      Object {
        "enforce": "pre",
        "test": /\\\\\\.tsx\\?\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/tslint-loader/index.js",
          },
        ],
      },
      Object {
        "enforce": "pre",
        "test": /\\\\\\.js\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/source-map-loader/index.js",
          },
        ],
      },
    ],
  },
  "plugins": Array [
    CheckerPlugin {},
  ],
}
`;

exports[`valid preset production 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\\\\\.tsx\\?\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/awesome-typescript-loader/dist/entry.js",
          },
        ],
      },
      Object {
        "enforce": "pre",
        "test": /\\\\\\.tsx\\?\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/tslint-loader/index.js",
          },
        ],
      },
      Object {
        "enforce": "pre",
        "test": /\\\\\\.js\\$/,
        "use": Array [
          Object {
            "loader": "/projects/neutrino-middlewares/node_modules/source-map-loader/index.js",
          },
        ],
      },
    ],
  },
}
`;
